backend/battling/views.py

```diff
@@ -15,7 +16,7 @@ class Home(TemplateView):
     template_name = "battling/home.html"
 
 
-class CreateBattle(CreateView):
+class CreateBattle(LoginRequiredMixin, CreateView):
```

Comment:
Cool. Did you check ALL your views to see if ALL of them have `LoginRequiredMixin` when necessary?

Reply:
Yes. The user can only access the Home, Login and Signup views :D